### PR TITLE
Don't be lenient in PluginService#processModule(Module)

### PR DESCRIPTION
**Pull request by s1monw**
We today catch the exception and move on - this is no good, we should just fail
all the way if something can't be loaded or processed

### CODE REVIEW

**Comment by rmuir** [comment]
+1

**Comment by jaymode** [comment]
+1 on failing here. LGTM!

File: core/src/test/java/org/elasticsearch/plugins/PluginsServiceTests.java [hunk 1]
```diff
@@ -86,4 +110,17 @@ public void testAdditionalSettingsClash() {
             assertTrue(msg, msg.contains("plugin [additional-settings2]"));
         }
     }
+
+    public void testOnModuleExceptionsArePropergated() {
```
Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/plugins/PluginsServiceTests.java]: Nit: `testOnModuleExceptionsArePropergated` -> `testOnModuleExceptionsArePropagated`

**Comment by jasontedor** [comment]
I left a nitty comment about a spelling error but otherwise LGTM. :)

**Comment by rjernst** [comment]
LGTM too

**Comment by s1monw** [comment]
@jaymode @rmuir @rjernst @jasontedor i pushed some fixes - of course a test failed

**Comment by rjernst** [comment]
Still LGTM

**Pull request merged**